Lab3.2/KsiazkaZOkladkaZwykla.java

```diff
@@ -0,0 +1,36 @@
+public class KsiazkaZOkladkaZwykla implements Publikacja {
+
+    private Publikacja publikacja;
+
+    public KsiazkaZOkladkaZwykla(Publikacja publikacja){
+        if(publikacja instanceof KsiazkaZOkladkaTwarda || publikacja instanceof KsiazkaZOkladkaZwykla){
+            System.out.println("Książka nie może mieć dwóch okładek");
+        }
+        else{
+            this.publikacja = publikacja;
+        }
+    }
+
+    @Override
+    public String dajAutora(){
+        if(this.publikacja == null) return null;

```

Comment:
@repcus Nie są konieczne takie zabezpieczenia, jeśli ktoś przekazał nullową referencję do dekoracji to jego problem....